components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java

```diff
@@ -86,6 +89,23 @@ public Optional<User> resolveUserFromResidentOrganization(String userName, Strin
                             user = userStoreManager.getUser(null, userName);
                         } else if (userId != null && userStoreManager.isExistingUserWithID(userId)) {
                             user = userStoreManager.getUser(userId, null);
+                        } else if (userName != null) {
+                            boolean userFound = false;
+                            UserStoreManager secondaryUserStoreManager =
+                                    userStoreManager.getSecondaryUserStoreManager();
+                            while (secondaryUserStoreManager != null) {
+                                domain = secondaryUserStoreManager.getRealmConfiguration().getUserStoreProperties()
+                                        .get(PROPERTY_DOMAIN_NAME);
+                                if (userStoreManager.isExistingUser(domain + DOMAIN_SEPARATOR + userName)) {
```

Comment:
We should be able to directly invoke the isExistingUser method of the secondary user store manager , rather than using the user store domain

Reply:
The secondary user store manager is type of UserStoreManager and that interface do have `isExistingUser` method but not a method to retrieve the user like `getUser`.  In that case have to cast the UserStoreManager to the corresponding implementation at the runtime. The informative detail which can be extracted was the domain name from the corresponding realm configurations. 

With the domain name having in hand, can do the correct check from the primary user store manager also. These steps get executed when the username was not domain qualified.

Reply:
Answering for the question. Yes we can do that check, but that was not enough.